main.tf

```diff
@@ -11,7 +11,7 @@ terraform {
 }
 
 provider "azurerm" {
-  use_msi                    = var.use_msi_to_authenticate
+  use_msi                    = var.use_msi_to_authenticate || var.msi != null
```

Comment:
I believe we can take out this flag altogether using the ARM environment variables